django/contrib/admin/checks.py

```diff
@@ -109,6 +110,52 @@ def _check_raw_id_fields_item(self, obj, model, field_name, label):
             else:
                 return []
 
+    def _check_autocomplete_fields(self, obj):
+        """
+        Check that `autocomplete_fields` only contains field names that are
+        listed on the model.
+        """
+        if not isinstance(obj.autocomplete_fields, (list, tuple)):
+            return must_be('a list or tuple', option='autocomplete_fields', obj=obj, id='admin.E001')
+        else:
+            return list(chain(*[
+                self._check_autocomplete_fields_item(obj, obj.model, field_name, 'autocomplete_fields[%d]' % index)
+                for index, field_name in enumerate(obj.get_autocomplete_fields())
```

Comment:
I guess this branch isn't tested since `request` is a required argument and passed to `get_autocomplete_fields()`. Since we can't reasonable come up with a request in this context, maybe this check shouldn't happen in the method case.